pennylane/ops/arithmetic/sprod.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the SProd class which contains logic for
+computing the scalar product of operations.
+"""
+import pennylane as qml
+from pennylane import math
+from pennylane.operation import Operator, expand_matrix
+
+
+def s_prod(scalar, operator):
+    """Associate an operator with a scalar to represent scalar multiplication."""
+    return SProd(scalar, operator)
+
+
+class SProd(Operator):
+    """Arithmetic operator subclass representing the scalar product of an
+    operator with the given scalar."""
+
+    def __init__(self, scalar, operator, do_queue=True, id=None):
+        # Add validation checks for size / shape / type of scalar and operator
+
+        if isinstance(operator, self.__class__):  # SProd(0.5, SProd(3, PauliX)) = SProd(1.5, PauliX)
+            self.op = operator.op
+            self.scalar = scalar * operator.scalar
+        else:
+            self.op = operator
+            self.scalar = scalar
```

Comment:
Part of the motivation behind operator arithmetic is avoiding in-place updates.  If we want to eagerly combine coefficients, I would recommend moving that logic to the `sprod` constructor instead of having it the initialization method for the class.

Maybe something like 
```python
def sprod(scalar, operator, eager=True):
    if eager and isinstance(operator, SProd):
         return SProd(scalar * operator.scalar, operator.op)
    return SProd(scalar, operator.op)
```
That way we aren't dynamically updating the properties of a class in place.

This could also be the job of the simplification pipeline.

Reply:
I think pushing this logic into simplification pipeline is probably best!